pallets/randomness/src/vrf.rs

```diff
@@ -97,3 +99,14 @@ pub(crate) fn verify_and_set_output<T: Config>() {
 		RandomnessResults::<T>::insert(local_vrf_this_block, results);
 	}
 }
+
+pub(crate) fn set_previous_output<T: Config>(now: BlockNumberFor<T>) {
+	if let Some(randomness) = LocalVrfOutput::<T>::get() {
+		PreviousLocalVrfOutputs::<T>::mutate(|outputs| {
+			if outputs.len() > T::MaxPreviousLocalRandomness::get() as usize {
+				outputs.pop();
+			};
+			outputs.push((randomness.clone(), now));
+		});
+	}
```

Comment:
if we do this in `on-initialize` after the output is set in storage, then there is no need for this DB read

Also it would be better weight accounting to do it in on-initialize, possibly just inside `vrf::set_output` and then re-run benchmarks.